### ISSUE TITLE
Android - READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE are set to max sdk 28 by core

### ISSUE THREAD

**Issue by JWiseCoder** [NativeScript/NativeScript#10341]
### Issue Description

A recent update to core has caused both the READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE permissions to default to max sdk 28.  In order to fix this, I have to do the following to my app's AndroidManifest:

```
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" tools:replace="android:maxSdkVersion" android:maxSdkVersion="32" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" tools:remove="android:maxSdkVersion" />
```

I've never needed this before.

SDK 28 is not the correct version to limit these permissions to.  READ_EXTERNAL_STORAGE has a use from 29 up to and including 32, as indicated in the documentation:

[https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE)

Apps still need this permission in SDK 29-32 if they want to request access to items in the media store.  For sdk 33 and up, there are different permissions.

WRITE_EXTERNAL_STORAGE has no deprecation warning, and is needed for apps that write files outside the application-specific directories.

Can the `android:maxSdkVersion="28"` please be removed from that AndroidManifest in packages/ui-mobile-base/android/widgets/src/main, please?

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by insytes** [comment]
I have also run into this issue.

To fix I have similarly added these lines to the Android Manifest file

```xml
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="32" tools:replace="android:maxSdkVersion" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="32" tools:replace="android:maxSdkVersion" />

**Comment by JWiseCoder** [comment]
Yes, I mentioned that above.  Also, I think WRITE_EXTERNAL_STORE doesn't need to be limited to sdk 32.  It will still work on 33 and above.

**Comment by insytes** [comment]
@JWiseCoder Yes the WRITE_ is the issue here, as seen in https://github.com/NativeScript/android/blob/16abf31cf3993a75482fd4df9e000362a59ab7d5/test-app/app/src/main/AndroidManifest.xml#L5-L6

**Comment by JWiseCoder** [comment]
That's in the test app, but point taken.

I believe WRITE_EXTERNAL_STORAGE is implicit for files within the app's scope, but it (or another permission) is still needed for files outside that.  My point is that there is no sdk-related note on WRITE_EXTERNAL_STORAGE the same way there is for READ_EXTERNAL_STORAGE.

In any case, this should probably be left up to the app developer to determine, and should not be limited by nativescript core.

**Comment by insytes** [comment]
I was under the impression that the project template used when building the Android app comes from the @nativescript/android repository and within the test-app directory 🤔 

I agree with all your suggestions BTW.
